### ISSUE TITLE
 Error: OmniSharp server load timed out

### ISSUE THREAD

**Issue by wangzengqiao** [dotnet/vscode-csharp#4188]
Starting OmniSharp server at 2020/11/6 上午11:47:15
    Target: d:\cSharp\DotNetDebugging

OmniSharp server started.
    Path: c:\Users\wangzengqiao\.vscode\extensions\ms-dotnettools.csharp-1.23.5\.omnisharp\1.37.3\OmniSharp.exe
    PID: 110200

[ERROR] Error: OmniSharp server load timed out. Use the 'omnisharp.projectLoadTimeout' setting to override the default delay (one 
minute).

**Comment by tverboon** [comment]
You can add this: `"omnisharp.projectLoadTimeout": 600,` to your settings.json or change the OmniSharp settings via Visual Studio Code settings.

600 is the amount of seconds.

**Comment by gocs** [comment]
> You can add this: `"omnisharp.projectLoadTimeout": 600,` to your settings.json or change the OmniSharp settings via Visual Studio Code settings.
> 
> 600 is the amount of seconds.

n amount of seconds never loads the server at all. it just blocks there forever.

**Comment by oraad** [comment]
I'm having the same issue even after increasing the timeout for more than an hour. Is there a way to troubleshoot the reason it is timing out?

**Comment by oraad** [comment]
I uninstalled the extension and deleted the extension directory then re-installed the extension and it seems to have solved the problem

**Comment by nihilocrat** [comment]
I was also able to resolve this without changing the timeout by pressing Ctrl + Shift + P and choosing "OmniSharp: Restart Server", letting it time out and doing it again until it starts. I think it may be more prone to timing out if you have a big solution with a lot of projects in it. Not so bad if you keep the same project open all day.

**Comment by brnhensley** [comment]
I was just having this same issue and I had also uninstall the the extension and deleted the extension directory then re-installed the extension to fix it.

**Comment by Anamon** [comment]
I'm on a completely new and pristine system, just installed VS Code (1.67.1), the .NET SDK (6.0.300), and its C# extension (1.24.4). I created a new minimal project, the standard .NET console template (Hello world), and I'm running into this issue. OmniSharp always times out, neither restarting VS Code, nor restarting the OmniSharp server, nor reinstalling the extension, nor increasing the timeout helps.

I actually tried using OmniSharp years ago, probably around 2017–2018, and also never managed to get it to even load. I can't remember if it was the same issue, but it's been five years now and I have yet to manage to start OmniSharp even once. Surely I can't be the only one having made that experience?

Is there at least a way to figure out what, specifically, it's timing out on?
